### DISCUSSION TITLE
Bench cpu cycles instead of time.

### DISCUSSION THREAD

**Discussion by Zibri** [official-stockfish/Stockfish#4929] [Q&A]
I modified uci.cpp to add a measure of cpu cycles using RDTSC (on x86).
To my surprise, the result differs in every run.

```
$ for a in {1..10};do sudo nice -n -20 ./stockfish bench 16 8 13 2>&1|grep cycles;done
Total CPU cycles : 6997616972
Total CPU cycles : 8023006440
Total CPU cycles : 7189384157
Total CPU cycles : 7435561686
Total CPU cycles : 7699870524
Total CPU cycles : 7815712580
Total CPU cycles : 8237661264
Total CPU cycles : 7761370360
Total CPU cycles : 7604879767
Total CPU cycles : 7954136805
```

Can someone explain me why?
Is there some randomization somewhere?

I wish to find a way to get always the same number (and then try out some optimizations)

**Comment by Zibri** [comment]
The same happens in a single thread:
```
$ for a in {1..10};do sudo nice -n -20 ./stockfish bench 16 1 13 2>&1|grep cycles;done
Total CPU cycles : 5711339587
Total CPU cycles : 5609718495
Total CPU cycles : 5634257784
Total CPU cycles : 5644653996
Total CPU cycles : 5642880644
Total CPU cycles : 5633798297
Total CPU cycles : 5675089812
Total CPU cycles : 5689815720
Total CPU cycles : 5687945342
Total CPU cycles : 5694980278
```

**Reply by PGG106** [reply]
multithreaded search is non deterministic by design, for single threaded search the (search) results are deterministic but i don't know enough about measuring cpu cycles to weigh in.

**Reply by vondele** [reply]
there is no reason to expect the cycles to be identical, the OS might schedule on different cores, other processes or IO might interrupt, caches invalidated, etc. So, not a good idea to do this.
